### PR TITLE
hv: fix possible buffer overflow issues

### PR DESCRIPTION
**Pull request by yonghuah**
- cpu_secondary_init() @cpu.c
 - ptirq_intx_pin_remap() @ assign.c
   etc.

Tracked-On: #1252
Signed-off-by: Yonghua Huang <yonghua.huang@intel.com>
Acked-by: Eddie Dong <eddie.dong@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
2061<br>6 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/arch/x86/assign.c&nbsp;&nbsp;function:ptirq_intx_pin_remap&nbsp;&nbsp;offset:632:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure contains UR data flow anomalies. : entry<br><br>2.filename:/hypervisor/arch/x86/assign.c&nbsp;&nbsp;function:ptirq_intx_pin_remap&nbsp;&nbsp;offset:724:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Attempt to use uninitialised pointer. : entry<br><br>3.filename:/hypervisor/common/hypercall.c&nbsp;&nbsp;function:hcall_vm_intr_monitor&nbsp;&nbsp;offset:1066:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Value is not of appropriate type. :  (signed char and unsigned char): intr_hdr -> buf_cnt <= ( 24   * 2U<br><br>4.filename:/hypervisor/common/hypercall.c&nbsp;&nbsp;function:hcall_vm_intr_monitor&nbsp;&nbsp;offset:1066:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Literal value requires a U suffix.           : 24<br><br>5.filename:/hypervisor/common/hypercall.c&nbsp;&nbsp;function:hcall_vm_intr_monitor&nbsp;&nbsp;offset:1066:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Use of mixed mode arithmetic. :  (unsigned int and signed char): intr_hdr -> buf_cnt <= ( 24 * 2U<br><br>6.filename:/hypervisor/common/hypercall.c&nbsp;&nbsp;function:hcall_vm_intr_monitor&nbsp;&nbsp;offset:1066:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Value is not of appropriate type. :  (unsigned int and signed char): intr_hdr -> buf_cnt <= (   24 *<br><br>

File: hypervisor/arch/x86/assign.c [hunk 1]
```diff
@@ -644,83 +645,90 @@ int32_t ptirq_intx_pin_remap(struct acrn_vm *vm, uint8_t virt_pin,
 	 * entry already be held by others, return error.
 	 */
 
+	if (pic_pin && (virt_pin > NR_VPIC_PINS_TOTAL)) {
+		status = -EINVAL;
+	}
+
```
Comment (dongyaozu) [inline hypervisor/arch/x86/assign.c]: you can have "else" here, which is easier.

File: hypervisor/arch/x86/assign.c [hunk 2]
```diff
@@ -644,83 +645,90 @@ int32_t ptirq_intx_pin_remap(struct acrn_vm *vm, uint8_t virt_pin,
...
+					if (pic_pin) {
+						phys_pin = pic_ioapic_pin_map[virt_pin];
+					}
+					entry = add_intx_remapping(vm,
+							virt_pin, phys_pin, pic_pin);
+					if (entry == NULL) {
+						pr_err("%s, add intx remapping failed",
+								__func__);
+						status = -ENODEV;
+						spinlock_release(&ptdev_lock);
```
Comment (dongyaozu) [inline hypervisor/arch/x86/assign.c]: the spinlock can be moved out of the {}.  We can merge this now, without goto END

**Comment by HuiHuangShi** [comment]
2061<br>6 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/arch/x86/assign.c&nbsp;&nbsp;function:ptirq_intx_pin_remap&nbsp;&nbsp;offset:632:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Procedure contains UR data flow anomalies. : entry<br><br>2.filename:/hypervisor/arch/x86/assign.c&nbsp;&nbsp;function:ptirq_intx_pin_remap&nbsp;&nbsp;offset:724:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Attempt to use uninitialised pointer. : entry<br><br>3.filename:/hypervisor/common/hypercall.c&nbsp;&nbsp;function:hcall_vm_intr_monitor&nbsp;&nbsp;offset:1066:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Value is not of appropriate type. :  (signed char and unsigned char): intr_hdr -> buf_cnt <= ( 24   * 2U<br><br>4.filename:/hypervisor/common/hypercall.c&nbsp;&nbsp;function:hcall_vm_intr_monitor&nbsp;&nbsp;offset:1066:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Literal value requires a U suffix.           : 24<br><br>5.filename:/hypervisor/common/hypercall.c&nbsp;&nbsp;function:hcall_vm_intr_monitor&nbsp;&nbsp;offset:1066:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Use of mixed mode arithmetic. :  (unsigned int and signed char): intr_hdr -> buf_cnt <= ( 24 * 2U<br><br>6.filename:/hypervisor/common/hypercall.c&nbsp;&nbsp;function:hcall_vm_intr_monitor&nbsp;&nbsp;offset:1066:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Value is not of appropriate type. :  (unsigned int and signed char): intr_hdr -> buf_cnt <= (   24 *<br><br>

**Comment by HuiHuangShi** [comment]
2061<br>No new violations to the coding guideline detected.

**Comment by HuiHuangShi** [comment]
2061<br>No new violations to the coding guideline detected.

File: hypervisor/arch/x86/cpu.c [hunk 3]
```diff
@@ -412,6 +412,7 @@ void init_cpu_pre(uint16_t pcpu_id)
 	bitmap_set_nolock(pcpu_id, &pcpu_active_bitmap);
 
 	/* Set state for this CPU to initializing */
+	ASSERT(pcpu_id < CONFIG_MAX_PCPU_NUM, "");
 	cpu_set_current_state(pcpu_id, PCPU_STATE_INITIALIZING);
```
Comment (dongyaozu) [inline hypervisor/arch/x86/cpu.c]: Why we need this one?  The caller side is using fixed #

**Comment by yonghuah** [comment]
@dongyaozu  Eddie, have update for validating pCPU ID statement, pls review. thanks.

**Comment by acrnsi** [comment]
Ok to verify

**Comment by HuiHuangShi** [comment]
2061<br>No new violations to the coding guideline detected.

**Comment by HuiHuangShi** [comment]
2061<br>No new violations to the coding guideline detected.

**Pull request merged**